### ISSUE TITLE
Sprint - Sept 30 to Oct 11

### ISSUE THREAD

**Issue by Phanatic** [PostHog/posthog#25181]
# Global Sprint Planning

## 3 things that might take us down

1. Celery - not well detected, took us a while to figure out what is broken
2. 
3.

## Urgent incident follow-ups

https://github.com/orgs/PostHog/projects/103/views/2

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by pauldambra** [comment]
# Team No-Longer-Error

**Support hero:** @marandaneto    

TLDR
One week then an onboarding

## Retro

* delayed till David not OOO

### High priority

WEEK 1 ✅
- Get Richard up and running
- Get @oliverb123  up and running
- Get Q4 planning done

WEEK 2 

#### Mobile ✅ 

* RN replay - publishing should still be done by 16th and then we get people testing @marandaneto 
* start investigating flutter @marandaneto 

#### Error tracking

moved to new team and lots of planning done

#### Session replay

- ✅ 🟡 unplayable recordings @pauldambra 
    - lots of good feedback
    - fewer support tickets 	

### Low priority / side quests

- uncaptured network requests
   -  pushed to Q4 goal
- 🟡 error assignment into activity log @pauldambra 
- ✅ step based debug playback @daibhin 

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

🟡📱Goal 1: People think of PostHog as a mobile solution
🟡 📹 Goal 2: Replay playback 3000
    - delayed-ish till Nov and Cory time in Valencia
🟡 🎤 Goal3: Ship the things users keep asking for
 
### High priority

WEEK 1

## Mobile

* start implementing the flutter session replay plugin - lots can be re-used

## Replay

* start recording on a trigger (e.g. error captured) @richard-better 
* track impact of replay capture improvements @pauldambra 

WEEK 2

* onboard Yiannis
* nobody gets lost or arrested in Vienna

### Low priority / side quests

- uncaptured network requests @pauldambra 
- error assignment into activity log @pauldambra 
- experiment with SwiftUI masking @marandaneto 
   - a hacky workaround but should work and helps folk
   - need to discover performance impact

**Comment by benjackwhite** [comment]
# Team Creating Different Problems (CDP)
Support: @mariusandra 

Olly 50% or so focus on Exceptions
Last sprint for Ben before long vacation 🌴 🗻 😍 
Marcus also out 50%

## Retro

- General discussion around how to release this and being a little shackled by the existing pricing structure
- Options are:
  - Use the existing pricing (which sucks for low volume users)

- [ ] 🟡  Goal - quiet launch of the new CDP destinations
  - [x] Everything running on Cyclotron ensuring alerting and metrics etc. @benjackwhite
  - [x] All destinations ready to migrated @mariusandra @MarconLP 
  - [ ] Take it out of feature preview so that no new destinations are made as plugins
- [ ] 🔴 Slow burner getting more info about capture rust so we can finally get rid of django capture
- [ ] 🟢  Exception ingestion PoC with team replay


## OKRs
* Widespread usage of the new CDP 
* Messaging V1 
* Ingestion is rock solid

### High priority

- Communicate action and destination upgrades @mariusandra 
  - [ ] Plan out communication with users for the migration (especially in case things could go wrong)
  - [ ] One click option for upgrading an action webhook
  - [ ] Send a warning to customers that we are migrating them over with clear deadline (e.g. November)
- Bring destinations 3000 out of feature preview @benjackwhite 
  - Gradually move all teams over to cyclotron
  - Docs
  - Any other final hanging things to feel confident
- Documenting and updating runbooks for realtime destinations @benjackwhite 
- (Olly 60%(ish) energy on Exceptions ingestion)
- Pricing? @annaszell

**Comment by EDsCODE** [comment]
# Team Data House
### OKR Q2 2024

### Objective

**Query 3000**

* **Key Results:**
    * Autocomplete
	* Increase general BI experience/product https://github.com/PostHog/meta/issues/157
	* Declutter the data warehouse UI and make the features intuitive to find

**Data Modeling MVP**

* **Key Results:**
	* Infrastructure decided and implemented
	* Integrating external data with feature flags
	* External data everywhere in insights/persons/cohorts
	* Get billing team to use modeling in posthog for their invoices_with_annual table
 
### Retro

- [ ] Data Modeling MVP beta shipped (by end of this week)
- [ ] data warehouse data in funnels (+ rest of insights)
- [ ] more of this https://github.com/PostHog/meta/issues/157
- [ ] dashboard filters, links in columns
- [ ] insight variables (WIP)

### Priority

- [ ] dashboard insight variables @Gilbert09 
- [ ] data modeling Beta @tomasfarias 
	- [ ] UI refactoring (making materialized vs regular views really clear when querying, QA canvas view) @EDsCODE 
- [ ] salesforce custom fields (internal request) @EDsCODE

**Comment by daibhin** [comment]
# Team Error Tracking

tl;dr
@oliverb123 working with Error Tracking on the ingestion side of things

**Support hero:** 👀

## OKR

1. Human readable stack traces for Javascript
2. Ensure scalability of query time issue grouping
3. Alerting via CDP
4. Launch

### High priority

- Symbolify a single stacktrace @oliverb123 
  - Setup local environments
  - Spin up new Rust service
  - Configure deployments
  - Upload sourcemaps to S3 bucket @daibhin
  - Start exploring Sentry's symbolifier package

- Better filtering @neilkakkar
  - Finish exception content search
  - Filter by url, file, or task name
  - Add property to filter by source (projects concept in Sentry)

- Person overrides design planning @daibhin 
  - Catch up with Brett / Ted to understand person overrides table

### Low priority / side quests

- Enable autocapture on billing repo @daibhin
- Rename stack trace properties on JS SDK  (beware: Sentry integration relies on some of this) @neilkakkar

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>  
<summary>Retro items</summary>

- @zlwaterfield 
	- [x] Get annual credits based fully rolled out to first users
	- [x] Iterate on annual credits based on usage / feedback
	- [x] Annual credits monthly distributions
	- [x] RBAC PR hand off from Ben
	- [ ] Continue ICP analysis with new data from Mine
	- [x] Confirm customer.io integration is now working - work with Joe to get mailgun emails moved over
	- [x] Work with Joe to get existing startups onboarded to a campaign for roll out notifications
	- if time
		- [x] WorkOS splike for SAML and SCIM (limited to 1 day)
- @raquelmsmith 
    - [ ] Implement Personless default switch
    - [x] Organize work for https://github.com/PostHog/company-internal/issues/1452 with data science contractor
    - [x] Ship analytics onboarding templates
    - [ ] Don't delete all upcoming invoices at once when we sync them - so we don't see daily big drops in revenue for half the day
    - [ ] Move all activation metrics to backend (in progress)
</details>  

## Q3 Goals
✅=finished 🟡=in progress 🔴=won't finish ⚪=not started
1. 🟡 Make onboarding awesome for Product analytics and Data warehouse (Raquel)
2. 🟢  Support self-serve annual commitments (Zach)
3. 🟡 Dive into the data to understand our billing metrics and customers better (Zach)
4. ✅ Launch pricing for data warehouse (Raquel)
5. 🟢 Hire 2 people (one for billing, one for product/experimentation focus)

## This sprint

Time off: Offsite first week! 

- @zlwaterfield 
	- [ ] next round of emails from mailgun to c.io 
	- annual credits
		- [ ] release to more users
		- [ ] make the banner closable but still show on the page
		- [ ] https://posthog.slack.com/archives/C01MGUHFH6G/p1726790198011829 - credits ui for annual customer and existing customers with a discount
	- RBAC
		- [ ] Finish reviewing Ben's existing PR and RFC 
		- [ ] Finish v1 of the RFC for billing	
		- [ ] Start working on it - decide if we should continue on Ben's PR or start from scratch and do it piece by piece - get at least 1 PR merged for it
	- [ ] (from previous) ICP analysis - review completed work and improve charts/dashboard for new data flowing in from SalesForce
- @raquelmsmith 
    - [ ] Implement Personless default switch
    - [ ] Keep working with new onboarding templates flow
    - [ ]  Move activation events to the backend
    - [ ] Help Patricio get up and running
    - [ ] Work with data science contractor

- @patricio-posthog 
	- [x] Get local development running (Posthog and Billing)
	- [x] Get first PR pushed
	- [ ] Add emails/events for first "big bills" - customers are still getting surprised by big bills
	- [ ] Sync invoices - don't drop all upcoming, but keep them up to date with webhooks and expiration

**Comment by fuziontech** [comment]
# Team Where Do We Put Our Data? 
![image](https://github.com/user-attachments/assets/9b9485f0-2537-481a-9633-1ad17851b64b)

## OKR Q2 2024

### Objective

**Goals for Q4:**

- Improve elasticity and flexibility of our data store by putting all our data in Iceberg
    - Work with Altinity effectively to ship read path for Iceberg on ClickHouse - Brett Hoerner
    - Setup infrastructure to ship all of our data to Iceberg on S3 - James Greenhill
    - Shipping Query logs to Iceberg - James Greenhill
- Continue improving CH operational expertise
    - Upgrade to a later version of Clickhouse - James Greenhill
    - Capacity planning - James Greenhill
    - Automation - Daniel Escribano
    - Put some of the basic mitigation operations in runbooks - Daniel Escribano
- Schema management
    - Tool for schema migration (coordinator schemas) - Daniel Escribano
    - Tool for long running mutations - Daniel Escribano
- Continued investment in performance
    - Tooling for other teams to understand which queries are slow and why - Ted Kaemming
    - Investigate variability of queries - Ted Kaemming
    - Per-team limits on queries/query complexity (needs product work) - Ted Kaemming

## Board
https://github.com/orgs/PostHog/projects/85/views/2

## Retro

@tkaemming - Shipped all the US properties map stuff, looking really fast. Ted is a query now.

@Daesgar - Positive side - I think we are at a good level of automation. We finally have a good plan for clickhouse offline cluster which is struggling. Negative side - It took too much for us to have this quota upgrade. Need to invest time in Parquet.

@fuziontech - I'm feeling really good about the current direction, but I do think we need to go really really deep on how we want to have the data laid out on S3 (obviously) but this is a fun problem to have.

@bretthoerner - Feeling pretty excited about what we are doing in terms of the direction, but we need to get ahead of any issues we might not be thinking of.

## Board Snapshot 

TL;DR

- Write path for Iceberg setup to iterate
- Backup cluster is moved to i4i.metal (more disk more io)
- Upgraded to Altinity 24.3
- ^ these two unlock real lightweight deletes
- We have a plan with Altinity

<img width="1088" alt="image" src="https://github.com/user-attachments/assets/bf39798a-72ee-4826-969a-4d6b49dded52">

**Comment by benjackwhite** [comment]
# Team Infra

## Retro / hangover

- 🔴 Comply with blockage of IPs from Russia @frankh
  - 🟢  We now have a banner to warn
  - 🔴 We haven't enabled it and the actual blocking needs some thinking (would be good to pull in a FS engineer here)
- 🔴 PostHog JS - deploy its own assets 
- 🔴 Frank to focus on "infra sessions" for onboarding Bryan
- 🟢 ArgoCD @danielxnj @Daesgar
  - 🟢 have a full E2E CI/CD flow for an app
  - 🟡 One by one migrated all "internal" services
- 🟡 Make sure we do a great onboarding for Bryan 
  -  Lots of learning and crossover with peeps
  - Time to start shipping some things 😉 
- Secrets management testing
 
## OKR

1. 🏃 Continuous Delivery
2. 💰 Cost visualization
3. 🧪 Continuous Integration

### High Priority
- ArgoCD all the way @danielxnj @bciaraldi 
  - We want all internal k8s stuff deployed with ArgoCD
  - We want a PostHog service deployed with ArgoCD
- PoC'ing out what a great secret management workflow would look like @bciaraldi 

### Side quest
- Drift detection

**Comment by aspicer** [comment]
# Team Product Analytics [WIP]
## Last Week
Roll UDF-based funnels to all users @aspicer, with @skoob13
Ensure Product Analytics can deploy UDF updates e.g. via script


Rewrite cohorts to HogQL finally @aspicer, with @Twixes
 - Plan


Context for Q4 planning @annaszell
 - Done. Planning happened

Roll out analytics alerts to users @anirudhpillai

**Comment by robbie-c** [comment]
# Team Web Analytics
**Support hero:** @robbie-c 

## Retro

The marketing launch happened! The traffic went up!- Doc updates
- 🔴 Add web analytics to onboarding flow
- 🔴 Work with customers still on sessions v1 to figure out a timeline for delete it
- 🔴 Spike for whether we can speed up web analytics by querying all the breakdowns at once using clickhouse maps

I got pretty distracted this sprint. Some unplanned stuff I did
- 🟢 Improved our docs for the following:
  - Self-referral traffic
  - Session ID requirements
  - Differences between different analytics tools
- 🟢 Add custom events as conversion goals
- 🟢 Exclude subdomains in the external click query
- 🟡 WIP but a bugfix for when campaign props from different visits get merged, and measure the impact of that fix

I'm out 1 week of the next sprint (I moved the holiday that was meant to be a couple weeks ago but got cancelled).

## OKR

1. 🟡 Make querying fast enough for large customers
2. 🟢 Heavily requested features
3. 🟢 Work better with other products
4. 🔴  Product and growth

### High priority
- Add web analytics to onboarding flow
- Debug with @fuziontech the sessions table v2 on 24.3
- Spike for whether we can speed up web analytics by querying all the breakdowns at once using clickhouse maps
